### ISSUE TITLE
Migration tool: failing to convert nested content properly

### ISSUE THREAD

**Issue by Sibren** [umbraco/Umbraco-CMS#6030]
I have recently migrated my site from 7.13.1 to 7.14 and then 8.1.1.
The site relies heavily on nested content and that would be quite the hassle to copy all of that by hand.
But now, with most pages in the backoffice with nested content, I get an error: 

> Error converting value 310 to type 'System.String[]'. Path '', line 1, position 3.
>
>Exception Details
> Newtonsoft.Json.JsonSerializationException: Error converting value 310 to type 'System.String[]'. Path '', line 1, position 3.

And when I try to load them in my controller, it says it misses the guid.
Which is correct.

This is my data:
`{"properties":{"pageTitle":[{"culture":"","seg":"","val":"Diensten - Internetbureau W3S Rotterdam"}],"metaDescription":[{"culture":"","seg":"","val":"META!"}],"hidePageInSitemap":[{"culture":"","seg":"","val":0}],"sitemapPriority":[],"metaRobotsIndex":[{"culture":"","seg":"","val":"[\"Index\"]"}],"metaRobotsFollow":[],"headerScripts":[],"bodyScripts":[],"ShowInMenu":[{"culture":"","seg":"","val":1}],"blocks":[{"culture":"","seg":"","val":"[{\"name\":\"Block Text\",\"ncContentTypeAlias\":\"BlockText\",\"title\":\"Title\",\"text\":\"<p>Content.</p>\",\"cta1\":\"[{\\\"name\\\":\\\"CallToAction\\\",\\\"udi\\\":\\\"umb://document/83f35e62e131497596833f3391312003\\\"}]\",\"cta2\":\"[]\",\"backgroundColor\":\"\",\"showPadding\":1},{\"name\":\"Block Services\",\"ncContentTypeAlias\":\"BlockServices\",\"services\":\"[{\\\"name\\\":\\\"Service\\\",\\\"ncContentTypeAlias\\\":\\\"BlockServicesItem\\\",\\\"iconImage\\\":\\\"umb://media/52b094ddb3d341579b3eb5097fe1618d\\\",\\\"title\\\":\\\"Tittle\\\",\\\"image\\\":\\\"umb://media/f6e96ac647904b4f81e386e508452a8c\\\",\\\"link\\\":\\\"[{\\\\\\\"name\\\\\\\":\\\\\\\"Name\\\\\\\",\\\\\\\"udi\\\\\\\":\\\\\\\"umb://document/9f8e3197461841ee800c60c97cac887a\\\\\\\"}]\\\"},{\\\"name\\\":\\\"Service\\\",\\\"ncContentTypeAlias\\\":\\\"BlockServicesItem\\\",\\\"iconImage\\\":\\\"umb://media/9e4a915c682b444dbda520365619da3d\\\",\\\"title\\\":\\\"Title\\\",\\\"image\\\":\\\"umb://media/35f36e501c2b4ffbbdb7393a395787ff\\\",\\\"link\\\":\\\"[{\\\\\\\"name\\\\\\\":\\\\\\\"Name\\\\\\\",\\\\\\\"udi\\\\\\\":\\\\\\\"umb://document/03e06d6e363a434db49da2d40f821f5e\\\\\\\"}]\\\"},{\\\"name\\\":\\\"Service\\\",\\\"ncContentTypeAlias\\\":\\\"BlockServicesItem\\\",\\\"iconImage\\\":\\\"umb://media/9fde4e6ef3da41db82135c6f5786439c\\\",\\\"title\\\":\\\"Ttitle\\\",\\\"image\\\":\\\"umb://media/793c63d948a7447fadde9282786c2d0b\\\",\\\"link\\\":\\\"[{\\\\\\\"name\\\\\\\":\\\\\\\"Name\\\\\\\",\\\\\\\"udi\\\\\\\":\\\\\\\"umb://document/0cfb89fb25a046ddb53387239fcdf5c2\\\\\\\"}]\\\"},{\\\"name\\\":\\\"Service\\\",\\\"ncContentTypeAlias\\\":\\\"BlockServicesItem\\\",\\\"iconImage\\\":\\\"umb://media/04f8841438944ea8956a136903f8cdae\\\",\\\"title\\\":\\\"Title\\\",\\\"image\\\":\\\"umb://media/90d214ee301348a78ffebca6628e1ed8\\\",\\\"link\\\":\\\"[{\\\\\\\"name\\\\\\\":\\\\\\\"Name\\\\\\\",\\\\\\\"udi\\\\\\\":\\\\\\\"umb://document/9df4391d398f49b0a1e121b260787dac\\\\\\\"}]\\\"}]\"},{\"name\":\"Block Contact\",\"ncContentTypeAlias\":\"BlockContact\",\"title\":\"Title!\",\"description\":\"<p>ContactUsBlabla.</p>\",\"useTabStyle\":null,\"showContactForm\":\"310\",\"showCallMeBackForm\":1,\"contactFormText\":\"\",\"callbackFormText\":\"\"}]"}]},"cultureData":{},"urlSegment":"diensten"}`

The 310 is a value from a custom dropdown.

Reproduction
------------
Create custom dropdown in 7.14 and use it in nested content block.
Upgrade.
Block is now broken.

Node is now broken in backoffice and backend.
It seems there's no guid set and that Umbraco.Dropdown became Umbraco.Dropdown.Flexible, which allows multiple items, so it expects that.

**Comment by stevemegson** [comment]
The dropdowns issue will be fixed by PR #6002. I hadn't encountered the problem with missing GUIDs, but I should be able to update that PR to deal with it.

**Comment by nul800sebastiaan** [comment]
@stevemegson I think you added the functionality to fix this a few days ago as well right? https://github.com/umbraco/Umbraco-CMS/pull/6002/commits/9acb79c7404f2996ebfcbce8028d8f08c07b7f6a

**Comment by stevemegson** [comment]
@nul800sebastiaan Yes, I thought I'd commented again to say that it was done, but clearly I imagined that.

**Comment by Zeegaan** [comment]
I will close this issue, please read https://our.umbraco.com/Documentation/Getting-Started/Setup/Upgrading/migrating-to-v8#known-issues for further instructions on how to go about migrating 👍
